server/src/main/java/org/opensearch/index/translog/InternalTranslogManager.java

```diff
@@ -83,11 +84,14 @@ public InternalTranslogManager(
      * Rolls the translog generation and cleans unneeded.
      */
     @Override
-    public void rollTranslogGeneration() throws TranslogException {
+    public void rollTranslogGeneration() throws TranslogException, IOException {
         try (ReleasableLock ignored = readLock.acquire()) {
             engineLifeCycleAware.ensureOpen();
             translog.rollGeneration();
             translog.trimUnreferencedReaders();
+        } catch (TranslogUploadFailedException e) {
+            // Do not trigger the translogEventListener as it fails the Engine while this is only an issue with remote upload
+            throw e;
```

Comment:
why catch when we need to throw?

Reply:
By default, the behavior is to catch the exception and then fail the shard and then throw the exception back to the caller. By catching and throwing, we are skipping the part of failing the shard here.